src/amo/components/RatingManager/index.js

```diff
@@ -175,47 +206,77 @@ export class RatingManagerBase extends React.Component<InternalProps, State> {
     );
   }
 
-  onReviewSubmitted = () => {
-    this.setState({ showTextEntry: false });
-    if (this.props.onReviewSubmitted) {
-      this.props.onReviewSubmitted();
+  renderTextEntry() {
+    const { _config, userReview } = this.props;
+    invariant(userReview, 'userReview is required');
+
+    if (_config.get('enableInlineAddonReview')) {
+      return <AddonReviewManager review={userReview} />;
     }
-  };
 
-  render() {
-    const { i18n, addon, userId, userReview, version } = this.props;
-    const { showTextEntry } = this.state;
-    const isLoggedIn = Boolean(userId);
+    return (
+      <AddonReview
+        onReviewSubmitted={this.onReviewSubmitted}
+        review={userReview}
+      />
+    );
+  }
 
-    invariant(addon, 'addon is required');
-    invariant(version, 'version is required');
+  renderUserRatingForm() {
+    const { i18n, addon, userReview } = this.props;
 
     const prompt = i18n.sprintf(
       i18n.gettext('How are you enjoying your experience with %(addonName)s?'),
       { addonName: addon.name },
     );
 
+    return (
+      <form action="">
+        <fieldset>
+          <legend className="RatingManager-legend">{prompt}</legend>
+          <div className="RatingManager-ratingControl">
+            {!this.isSignedIn() ? this.renderLogInToRate() : null}
+            <UserRating
+              readOnly={!this.isSignedIn()}
+              onSelectRating={this.onSelectRating}
+              review={!this.isSignedIn() ? null : userReview}
+            />
+          </div>
+        </fieldset>
+      </form>
+    );
+  }
+
+  renderInlineReviewControls() {
+    const { i18n } = this.props;
+
+    return this.shouldShowTextEntry() ? (
+      <Button
+        href="#cancelTextEntry"
+        onClick={this.cancelTextEntry}
+        className="RatingManager-cancelTextEntryButton"
+        buttonType="neutral"
+        puffy
+      >
+        {i18n.gettext("Nevermind, I don't want to write a review")}
```

Comment:
this might be more of a UX question, but if the use already has a review, maybe this text should change to indicate that? Otherwise maybe it's unclear that it won't delete the review (if it exists already)..?

Reply:
I totally missed that -- thanks! I will fix it.